backend/src/main/scala/cromwell/backend/async/ExecutionHandle.scala

```diff
@@ -27,19 +28,32 @@ final case class PendingExecutionHandle[BackendJobId <: JobId, BackendRunInfo, B
   override val result = NonRetryableExecution(new IllegalStateException("PendingExecutionHandle cannot yield a result"))
 }
 
-final case class SuccessfulExecutionHandle(outputs: CallOutputs, returnCode: Int, jobDetritusFiles: Map[String, Path], executionEvents: Seq[ExecutionEvent], resultsClonedFrom: Option[BackendJobDescriptor] = None) extends ExecutionHandle {
+final case class SuccessfulExecutionHandle(outputs: CallOutputs,
+                                           returnCode: Int,
+                                           jobDetritusFiles: Map[String, Path],
+                                           executionEvents: Seq[ExecutionEvent],
+                                           resultsClonedFrom: Option[BackendJobDescriptor] = None) extends ExecutionHandle {
   override val isDone = true
   override val result = SuccessfulExecution(outputs, returnCode, jobDetritusFiles, executionEvents, resultsClonedFrom)
 }
 
-final case class FailedNonRetryableExecutionHandle(throwable: Throwable, returnCode: Option[Int] = None) extends ExecutionHandle {
+sealed trait FailedExecutionHandle extends ExecutionHandle {
+  def kvPairsToSave: Option[Seq[KvPair]]
+}
+
+final case class FailedNonRetryableExecutionHandle(throwable: Throwable,
+                                                   returnCode: Option[Int] = None,
+                                                   override val kvPairsToSave: Option[Seq[KvPair]] = None) extends FailedExecutionHandle {
```

Comment:
TOL: I'd probably not provide a default here. I kind of like the idea of forcing creators of these handles to explicitly opt-out of storing key/value pairs